packages/hooks-react/src/usePlaylist.ts

```diff
@@ -1,42 +1,84 @@
+import { PLAYLIST_TYPE } from '@jwp/ott-common/src/constants';
+import { useConfigStore } from '@jwp/ott-common/src/stores/ConfigStore';
 
 const placeholderData = generatePlaylistPlaceholder(30);
 
-export default function usePlaylist(playlistId?: string, params: GetPlaylistParams = {}, enabled: boolean = true, usePlaceholderData: boolean = true) {
+export const getPlaylistQueryOptions = ({
+  type,
+  contentId,
+  siteId,
+  enabled,
+  usePlaceholderData,
+  params = {},
+  queryClient,
+}: {
+  type: PlaylistMenuType;
+  contentId: string | undefined;
+  siteId: string;
+  enabled: boolean;
+  queryClient: QueryClient;
+  usePlaceholderData?: boolean;
+  params?: GetPlaylistParams;
+}) => {
   const apiService = getModule(ApiService);
-  const queryClient = useQueryClient();
-
-  const callback = async (playlistId?: string, params?: GetPlaylistParams) => {
-    const playlist = await apiService.getPlaylistById(playlistId, { ...params });
 
-    // This pre-caches all playlist items and makes navigating a lot faster.
-    playlist?.playlist?.forEach((playlistItem) => {
-      queryClient.setQueryData(['media', playlistItem.mediaid], playlistItem);
-    });
+  return {
+    enabled: !!contentId && enabled,
+    queryKey: ['playlist', type, contentId],
+    queryFn: async () => {
+      if (type === PLAYLIST_TYPE.playlist) {
+        const playlist = await apiService.getPlaylistById(contentId, { page_limit: params?.page_limit });
 
-    return playlist;
-  };
+        // This pre-caches all playlist items and makes navigating a lot faster.
+        playlist?.playlist?.forEach((playlistItem) => {
+          queryClient.setQueryData(['media', playlistItem.mediaid], playlistItem);
+        });
 
-  const queryKey = ['playlist', playlistId, params];
-  const isEnabled = !!playlistId && enabled;
+        return playlist;
+      } else if (type === PLAYLIST_TYPE.content_list) {
+        const contentList = await apiService.getContentList({ siteId, id: contentId });
 
-  return useQuery<Playlist | undefined, ApiError>(queryKey, () => callback(playlistId, params), {
-    enabled: isEnabled,
-    placeholderData: usePlaceholderData && isEnabled ? placeholderData : undefined,
-    refetchInterval: (data, _) => {
+        return contentList;
+      }
+    },
+    placeholderData: contentId && usePlaceholderData ? placeholderData : undefined,
+    refetchInterval: (data: Playlist | undefined) => {
       if (!data) return false;
 
       const autoRefetch = isTruthyCustomParamValue(data.refetch) || data.playlist.some(isScheduledOrLiveMedia);
 
       return autoRefetch ? 1000 * 30 : false;
     },
     retry: false,
-  });
+  };
+};
+
+export default function usePlaylist({
+  contentId,
+  type,
+  params = {},
+  enabled = true,
+  usePlaceholderData = true,
+}: {
+  contentId?: string;
+  type: PlaylistMenuType;
+  params?: GetPlaylistParams;
+  enabled?: boolean;
+  usePlaceholderData?: boolean;
```

Comment:
I'm now seeing @dbudzins feedback. Although I agree to that, this change will result in many changes and could be considered a breaking change. I would very much like to prevent that, but it's not a blocking issue.

Reply:
Breaking change is a valid point here. Maybe we can start with hooks here and refactor usePlaylist / usePlaylists later.

Reply:
We consume these hooks in all platforms. The breaking change is that the hook signature changes. It would be safe if it was:

```ts
export default function usePlaylist(playlistId?: string, params: GetPlaylistParams = {}, enabled: boolean = true, usePlaceholderData: boolean = true, type: PlaylistMenuType = PLAYLIST_TYPE.playlist) {
```

Although this require adding all optional arguments with the default values when only changing the type.

It's not a huge deal, but something to be aware of.